File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var result: [Int] = []
    var progresses = progresses
    var speeds = speeds
    
    while !progresses.isEmpty {
        // 현재 진행 속도를 더함
        for i in 0..<progresses.count {
            progresses[i] += speeds[i]
        }
        
        var count: Int = 0
        // 완료된 태스크는 넘는건 제거
        while progresses.count > 0 && progresses[0] >= 100 {
            count += 1
            progresses.removeFirst()
            speeds.removeFirst()
        }
        
        if count > 0 {
            result.append(count)
        }

    }
    
    return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let remainDays = []
    progresses.forEach((progress, index) => {
        const remainProgress = 100 - progress
        const speed = speeds[index]
        remainDays.push(Math.ceil(remainProgress / speed))
    })
    
    // 지나간 일
    let currentDay = remainDays[0]
    let answer = [0]
    for (const remainDay of remainDays) {
        // 앞 작업보다 빨리 끝났다면
        if (remainDay <= currentDay) {
            answer[answer.length - 1] += 1
        } else {
            currentDay = remainDay
            answer.push(1)
        }
    }
    
    
    return answer;
}